### ISSUE TITLE
Compiling docker gives error

### ISSUE THREAD

**Issue by Kruszylo** [VROOM-Project/vroom#257]
Hi, I'm trying to build docker image with your project and something goes wrong, can you please check the error:
```
...
Processing triggers for sgml-base (1.26+nmu4ubuntu1) ...
Cloning into 'vroom'...
g++ -MMD -MP -I. -std=c++14 -Wextra -Wpedantic -Wall -O3 -DNDEBUG -std=c++1y -DBOOST_TEST_DYN_LINK -DBOOST_SPIRIT_USE_PHOENIX_V3 -DBOOST_RESULT_OF_USE_DECLTYPE -DBOOST_FILESYSTEM_NO_DEPRECATED -I/usr/include/lua5.2 -I/usr/local/include -I/usr/local/include/osrm -D USE_LIBOSRM=true -c main.cpp -o main.o
g++ -MMD -MP -I. -std=c++14 -Wextra -Wpedantic -Wall -O3 -DNDEBUG -std=c++1y -DBOOST_TEST_DYN_LINK -DBOOST_SPIRIT_USE_PHOENIX_V3 -DBOOST_RESULT_OF_USE_DECLTYPE -DBOOST_FILESYSTEM_NO_DEPRECATED -I/usr/include/lua5.2 -I/usr/local/include -I/usr/local/include/osrm -D USE_LIBOSRM=true -c algorithms/munkres.cpp -o algorithms/munkres.o
g++ -MMD -MP -I. -std=c++14 -Wextra -Wpedantic -Wall -O3 -DNDEBUG -std=c++1y -DBOOST_TEST_DYN_LINK -DBOOST_SPIRIT_USE_PHOENIX_V3 -DBOOST_RESULT_OF_USE_DECLTYPE -DBOOST_FILESYSTEM_NO_DEPRECATED -I/usr/include/lua5.2 -I/usr/local/include -I/usr/local/include/osrm -D USE_LIBOSRM=true -c algorithms/kruskal.cpp -o algorithms/kruskal.o
g++ -MMD -MP -I. -std=c++14 -Wextra -Wpedantic -Wall -O3 -DNDEBUG -std=c++1y -DBOOST_TEST_DYN_LINK -DBOOST_SPIRIT_USE_PHOENIX_V3 -DBOOST_RESULT_OF_USE_DECLTYPE -DBOOST_FILESYSTEM_NO_DEPRECATED -I/usr/include/lua5.2 -I/usr/local/include -I/usr/local/include/osrm -D USE_LIBOSRM=true -c algorithms/local_search/operator.cpp -o algorithms/local_search/operator.o
g++ -MMD -MP -I. -std=c++14 -Wextra -Wpedantic -Wall -O3 -DNDEBUG -std=c++1y -DBOOST_TEST_DYN_LINK -DBOOST_SPIRIT_USE_PHOENIX_V3 -DBOOST_RESULT_OF_USE_DECLTYPE -DBOOST_FILESYSTEM_NO_DEPRECATED -I/usr/include/lua5.2 -I/usr/local/include -I/usr/local/include/osrm -D USE_LIBOSRM=true -c algorithms/local_search/local_search.cpp -o algorithms/local_search/local_search.o
g++ -MMD -MP -I. -std=c++14 -Wextra -Wpedantic -Wall -O3 -DNDEBUG -std=c++1y -DBOOST_TEST_DYN_LINK -DBOOST_SPIRIT_USE_PHOENIX_V3 -DBOOST_RESULT_OF_USE_DECLTYPE -DBOOST_FILESYSTEM_NO_DEPRECATED -I/usr/include/lua5.2 -I/usr/local/include -I/usr/local/include/osrm -D USE_LIBOSRM=true -c algorithms/heuristics/solomon.cpp -o algorithms/heuristics/solomon.o
g++ -MMD -MP -I. -std=c++14 -Wextra -Wpedantic -Wall -O3 -DNDEBUG -std=c++1y -DBOOST_TEST_DYN_LINK -DBOOST_SPIRIT_USE_PHOENIX_V3 -DBOOST_RESULT_OF_USE_DECLTYPE -DBOOST_FILESYSTEM_NO_DEPRECATED -I/usr/include/lua5.2 -I/usr/local/include -I/usr/local/include/osrm -D USE_LIBOSRM=true -c algorithms/heuristics/clustering.cpp -o algorithms/heuristics/clustering.o
g++ -MMD -MP -I. -std=c++14 -Wextra -Wpedantic -Wall -O3 -DNDEBUG -std=c++1y -DBOOST_TEST_DYN_LINK -DBOOST_SPIRIT_USE_PHOENIX_V3 -DBOOST_RESULT_OF_USE_DECLTYPE -DBOOST_FILESYSTEM_NO_DEPRECATED -I/usr/include/lua5.2 -I/usr/local/include -I/usr/local/include/osrm -D USE_LIBOSRM=true -c routing/routed_wrapper.cpp -o routing/routed_wrapper.o
g++ -MMD -MP -I. -std=c++14 -Wextra -Wpedantic -Wall -O3 -DNDEBUG -std=c++1y -DBOOST_TEST_DYN_LINK -DBOOST_SPIRIT_USE_PHOENIX_V3 -DBOOST_RESULT_OF_USE_DECLTYPE -DBOOST_FILESYSTEM_NO_DEPRECATED -I/usr/include/lua5.2 -I/usr/local/include -I/usr/local/include/osrm -D USE_LIBOSRM=true -c routing/libosrm_wrapper.cpp -o routing/libosrm_wrapper.o
routing/libosrm_wrapper.cpp: In constructor 'vroom::routing::LibosrmWrapper::LibosrmWrapper(const string&)':
routing/libosrm_wrapper.cpp:38:18: error: no matching function for call to 'osrm::engine::EngineConfig::EngineConfig(<brace-enclosed initializer list>)'
     _osrm(_config) {
                  ^
In file included from /usr/local/include/osrm/engine_config.hpp:31:0,
                 from ./routing/libosrm_wrapper.h:13,
                 from routing/libosrm_wrapper.cpp:16:
/usr/local/include/osrm/engine/engine_config.hpp:73:8: note: candidate: osrm::engine::EngineConfig::EngineConfig()
 struct EngineConfig final
        ^
/usr/local/include/osrm/engine/engine_config.hpp:73:8: note:   candidate expects 0 arguments, 1 provided
/usr/local/include/osrm/engine/engine_config.hpp:73:8: note: candidate: osrm::engine::EngineConfig::EngineConfig(const osrm::engine::EngineConfig&)
/usr/local/include/osrm/engine/engine_config.hpp:73:8: note:   no known conversion for argument 1 from '<brace-enclosed initializer list>' to 'const osrm::engine::EngineConfig&'
/usr/local/include/osrm/engine/engine_config.hpp:73:8: note: candidate: osrm::engine::EngineConfig::EngineConfig(osrm::engine::EngineConfig&&)
/usr/local/include/osrm/engine/engine_config.hpp:73:8: note:   no known conversion for argument 1 from '<brace-enclosed initializer list>' to 'osrm::engine::EngineConfig&&'
makefile:52: recipe for target 'routing/libosrm_wrapper.o' failed
make: *** [routing/libosrm_wrapper.o] Error 1
The command '/bin/sh -c mkdir -p /src && 	cd /src && 	apt-get update && 	apt-get install -qq 		build-essential 		g++ 		git-core 	libboost-iostreams1.58-dev 		libboost-iostreams1.58.0 		libboost-log-dev 		libboost-log1.58.0 		libboost-regex-dev 		libboost-system-dev 		libboost-thread-dev 		libtbb-dev 		libtbb2 		make 		pkg-config && 	git clone --depth 10 --branch $VROOM_BRANCH https://github.com/VROOM-Project/vroom.git && 	mkdir -p /src/vroom/bin && 	cd /src/vroom/src && 	make && 	cp /src/vroom/bin/vroom /usr/local/bin && 	apt-get purge -qq 		'*-dev' 		build-essential 		git-core 		make && 	apt-get autoremove --purge -qq && 	apt-get clean && 	cd / && 	rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /src' returned a non-zero code: 2
```

My Docker file looks like:

```
FROM ubuntu:xenial
ENV OSRM_BACKEND_VERSION v5.10.0
RUN apt-get update && \
	apt-get install -qq \
		build-essential \
		cmake \
		git-core \
		libboost-all-dev \
		libboost-filesystem1.58.0 \
		libboost-iostreams1.58.0 \
		libboost-program-options1.58.0 \
		libboost-regex1.58.0 \
		libboost-system1.58.0 \
		libboost-thread1.58.0 \
		libbz2-dev \
		libexpat1 \
		libgomp1 \
		liblua5.2-dev \
		liblua5.2-0 \
		libluabind-dev \
		libluabind0.9.1v5 \
		libosmpbf-dev \
		libpng16-dev \
		libprotobuf-dev \
		libprotobuf9v5 \
		libprotoc-dev \
		libstxxl-dev \
		libstxxl1v5 \
		libtbb-dev \
		libtbb2 \
		libxml2-dev \
		libzip-dev \
		lua5.2 \
		luajit \
		pkg-config \
		protobuf-compiler && \
	mkdir -p /src && \
	cd /src && \
	git clone --depth 10 --branch $OSRM_BACKEND_VERSION https://github.com/Project-OSRM/osrm-backend && \
	cd osrm-backend && \
	mkdir -p build && \
	cd build && \
	cmake .. -DCMAKE_BUILD_TYPE=Release && \
	cmake --build . && \
	cmake --build . --target install && \
	mkdir -p /usr/local/share/osrm && \
	cp -a /src/osrm-backend/profiles /usr/local/share/osrm && \
	apt-get purge -qq \
		'*-dev' \
		build-essential \
		cmake \
		git-core \
		krb5-locales \
		make \
		wget && \
	apt-get autoremove --purge -qq && \
	apt-get clean && \
	cd / && \
	rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /src

ENV VROOM_BRANCH release/1.4
RUN mkdir -p /src && \
	cd /src && \
	apt-get update && \
	apt-get install -qq \
		build-essential \
		g++ \
		git-core \
		libboost-iostreams1.58-dev \
		libboost-iostreams1.58.0 \
		libboost-log-dev \
		libboost-log1.58.0 \
		libboost-regex-dev \
		libboost-system-dev \
		libboost-thread-dev \
		libtbb-dev \
		libtbb2 \
		make \
		pkg-config && \
	git clone --depth 10 --branch $VROOM_BRANCH https://github.com/VROOM-Project/vroom.git && \
	mkdir -p /src/vroom/bin && \
	cd /src/vroom/src && \
	make && \
	cp /src/vroom/bin/vroom /usr/local/bin && \
	apt-get purge -qq \
		'*-dev' \
		build-essential \
		git-core \
		make && \
	apt-get autoremove --purge -qq && \
	apt-get clean && \
	cd / && \
	rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /src

ENV VROOM_EXPRESS_BRANCH master
RUN apt-get update && \
	apt-get install -qq \
		build-essential \
		git-core \
		npm \
		nodejs-legacy && \
	cd / && \
	git clone --depth 10 --branch $VROOM_EXPRESS_BRANCH https://github.com/VROOM-Project/vroom-express.git && \
	cd vroom-express && \
	ln -s /dev/stdout access.log && \
	npm install && \
	apt-get purge -qq \
		build-essential \
		git-core \
		make  && \
	apt-get autoremove --purge -qq && \
	apt-get clean && \
	cd / && \
	rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /src

ENV VROOM_FRONTEND_BRANCH master
RUN apt-get update && \
	apt-get install -qq \
		build-essential \
		git-core \
		npm \
		nodejs-legacy && \
	cd / && \
	git clone --depth 10 --branch $VROOM_FRONTEND_BRANCH https://github.com/VROOM-Project/vroom-frontend.git && \
	cd vroom-frontend && \
	npm install && \
	cd /vroom-frontend/src && \
	make && \
	apt-get purge -qq \
		build-essential \
		git-core \
		make  && \
	apt-get autoremove --purge -qq && \
	apt-get clean && \
	cd / && \
	rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /src

# copied from postgres dockerfile
ENV GOSU_VERSION 1.10
RUN set -x \
	&& apt-get update && apt-get install -qq --no-install-recommends ca-certificates wget && rm -rf /var/lib/apt/lists/* \
	&& wget -O /usr/local/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$(dpkg --print-architecture)" \
	&& wget -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$(dpkg --print-architecture).asc" \
	&& export GNUPGHOME="$(mktemp -d)" \
	&& gpg --keyserver ha.pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 \
	&& gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu \
	&& rm -r "$GNUPGHOME" /usr/local/bin/gosu.asc \
	&& chmod +x /usr/local/bin/gosu \
	&& gosu nobody true

RUN apt-get update && \
	apt-get install -qq \
		curl \
		iproute  \
		iputils-ping \
		nginx \
		supervisor \
		wget && \
	apt-get clean && \
	cd / && \
	rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /src

COPY osrm.sh /usr/local/bin
COPY vroom-express.sh /usr/local/bin
COPY nginx.sh /usr/local/bin
COPY start.sh /usr/local/bin
COPY supervisord.conf /etc/supervisor/conf.d/supervisord.conf

VOLUME /osm

EXPOSE 5000
EXPOSE 3000

ENTRYPOINT ["/usr/local/bin/start.sh"]
```

**Comment by jcoupey** [comment]
There is no "official" docker image for `vroom`, can you share where yours comes from?

However the specific error you get:

```
routing/libosrm_wrapper.cpp: In constructor 'vroom::routing::LibosrmWrapper::LibosrmWrapper(const string&)':
routing/libosrm_wrapper.cpp:38:18: error: no matching function for call to 'osrm::engine::EngineConfig::EngineConfig(<brace-enclosed initializer list>)'
     _osrm(_config) {
```

sounds a lot like something related to https://github.com/VROOM-Project/vroom/issues/256#issuecomment-510439450. The proposed fix there is to use the `fix/libosrm-dataset-name` branch.

> ENV OSRM_BACKEND_VERSION v5.10.0

Maybe this ought to be updated with a more recent OSRM version.

**Comment by Kruszylo** [comment]
@jcoupey I found a docker image [crashbuggy/vroom-osrm
](https://hub.docker.com/r/crashbuggy/vroom-osrm), I wanted to use it but I. found out that it can't calculate routes for multiple vehicles due to old version of the codes, so I took Dockerfile from the page and to build a new image. At this moment I see that to make it compiled I need to update dependencies, some of them have a certain version (old version). 

I will change versions of libraries and will let you know the result. Thanks for the advice with the brunch, I also try to switch to it if with new libraries will not work.
